sdk/python/Dockerfile.conformance

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Dockerfile for building the source code of conformance tests
+FROM python:3.9.6-alpine3.14
```

Comment:
Do you have any reason to use Python 3.9 instead of Python 3.10?
We are supporting Python 3.10 in our Python SDK.

https://github.com/kubeflow/training-operator/blob/88719628eb7035cda66ea743c939a7d28599ad23/sdk/python/setup.py#L50

Reply:
No specific reason. I saw that we were using python 3.9 in a few places. Updated to use python 3.10 version. Let me know if I should use a different image.

Reply:
Thank you for updating it. I'm ok with `python:3.10-alpine` :)